src/fp.rs

```diff
@@ -36,16 +34,37 @@ pub(crate) struct FieldParameters {
 impl FieldParameters {
     /// Addition.
     pub fn add(&self, x: u128, y: u128) -> u128 {
-        let (z, carry) = x.wrapping_add(y).overflowing_sub(self.p2);
-        let m = 0u128.wrapping_sub(carry as u128);
-        z.wrapping_add(m & self.p2)
+        //   0,x
```

Comment:
nit: If these inline comments about how the arithmetic works are useful to readers, then I would like to see them added wherever appropriate (in particular, to `mul`). If they're not especially useful, then let's remove them.